bundles/http/Citrix/2ec16a1b-9423-45a6-8fd8-b0c42c916c0f/cb5694f4-3740-45d6-9817-44d451edb2d9.js

```diff
@@ -0,0 +1,1136 @@
+            if(key === 'next_page_token'){
+                user.next_page_token=value // 
+            }
+            else if(key === 'users'){
+                let usersData = value.map(userData =>{
+                    userContext.push(userData.id);
+                    let userObj = {
+                        "userid":userData.id,
+                        "first_name":userData.first_name,
+                        "last_name":userData.last_name,
+                        "email":userData.email,
+                        "timezone":userData.timezone,
+                        "last_login_time":moment.utc(userData.last_login_time).toDate()
+                    }
+                    userids.push(userObj); //storing the user data in array to be used in child endpoints
+                    return userObj;
+                })
+                dataStore.save("users",usersData);
+            }
+        });
+    }while(user.next_page_token!="");
+    context.user = userContext.toString();
+}
+
+//loading the meetings data (Upcoming Meetings)
+function syncMeetings(dataStore,client,userId=[]){
+    let i = 0;
+    let array = [];
+    //definig login to distinguish between fullsync and data update after action, as userId array is passed while performing data update after action
+    if(userId.length>0){
+       array = userId;
+    }
+    else{
+        array = userids;
+    }
+    do{
+        let response = client.fetchSync(`users/${array[i].userid}/meetings?type=upcoming&next_page_token=${meeting.next_page_token}&page_size=${meeting.page_size}`);
+        // console.log(`${userids[i]} status : ${response.status}`);
+        if(!response.ok){
+            throw new Error(response.jsonSync());
+        }
+        let responseData = response.jsonSync();
+        meeting.next_page_token = responseData['next_page_token'];
+        let meetings = responseData.meetings;
+        let data = meetings.map(meeting =>{
+            if(userId.length == 0){
+                meetingArray.push(meeting.id);
+            }
+            else{
+                meetingArray.push(array[i].meetingid)
+            }
+            return {
+                "uuid":meeting.uuid,
+                "id":Number(meeting.id),
+                "host_id":meeting.host_id,
+                "topic":meeting.topic,
+                "type":parseInt(meeting.type),
+                "start_time":moment.utc(meeting.start_time).toDate(),
+                "duration":parseInt(meeting.duration),
+                "timezone":meeting.timezone,
+                "created_at":moment.utc(meeting.created_at).toDate(),
+                "join_url":meeting.join_url,
+            }
+        });
+        dataStore.save('meetings',data);
+        if(meeting.next_page_token == ""){
+            i++;
+        }
+    }while(i<array.length);
+
+    //Removing the duplicate meetingids from the array
+    let meetingSet = new Set(meetingArray);
+    meetingArray = [...meetingSet];

```

Comment:
Why not use a Set from the start, and add elements to it directly? This will ensure that there are no duplicates without having to convert array -> set -> array.